### DISCUSSION TITLE
如何在 modal 中使用 affix?

### DISCUSSION THREAD

**Discussion by cdwmhcc** [arco-design/arco-design-vue#1374] [Q&A] [answered]
如何在 modal 中使用 affix?

**Comment by hehehai** [comment, marked as answer]
@cdwmhcc 你好，是像这样吗 [demo](https://arco-vue-playground.netlify.app/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcblxuY29uc3QgdmlzaWJsZSA9IHJlZihmYWxzZSlcbmNvbnN0IGNvbnRhaW5lclJlZiA9IHJlZigpXG5cbmNvbnN0IGhhbmRsZUNsaWNrID0gKCkgPT4ge1xuICB2aXNpYmxlLnZhbHVlID0gdHJ1ZVxufVxuY29uc3QgaGFuZGxlT2sgPSAoKSA9PiB7XG4gIHZpc2libGUudmFsdWUgPSBmYWxzZVxufVxuY29uc3QgaGFuZGxlQ2FuY2VsID0gKCkgPT4ge1xuICB2aXNpYmxlLnZhbHVlID0gZmFsc2Vcbn1cbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxhLWJ1dHRvbiBAY2xpY2s9XCJoYW5kbGVDbGlja1wiPk9wZW4gTW9kYWw8L2EtYnV0dG9uPlxuICA8YS1tb2RhbFxuICAgIHdpZHRoPVwiYXV0b1wiXG4gICAgdi1tb2RlbDp2aXNpYmxlPVwidmlzaWJsZVwiXG4gICAgQG9rPVwiaGFuZGxlT2tcIlxuICAgIEBjYW5jZWw9XCJoYW5kbGVDYW5jZWxcIlxuICA+XG4gICAgPHRlbXBsYXRlICN0aXRsZT4gVGl0bGUgPC90ZW1wbGF0ZT5cbiAgICA8ZGl2PlxuICAgICAgWW91IGNhbiBjdXNzdG9taXplIG1vZGFsIGJvZHkgdGV4dCBieSB0aGUgY3VycmVudCBzaXR1YXRpb24uIFRoaXMgbW9kYWxcbiAgICAgIHdpbGwgYmUgY2xvc2VkIGltbWVkaWF0ZWx5IG9uY2UgeW91IHByZXNzIHRoZSBPSyBidXR0b24uXG4gICAgICA8ZGl2IHN0eWxlPVwiaGVpZ2h0OiAyMDBweDsgb3ZlcmZsb3c6IGF1dG9cIiByZWY9XCJjb250YWluZXJSZWZcIj5cbiAgICAgICAgPGRpdiBzdHlsZT1cImhlaWdodDogNDAwcHg7IGJhY2tncm91bmQ6ICNjY2NjY2M7IG92ZXJmbG93OiBoaWRkZW5cIj5cbiAgICAgICAgICA8YS1hZmZpeCA6b2Zmc2V0VG9wPVwiMjBcIiA6dGFyZ2V0PVwiY29udGFpbmVyUmVmXCIgc3R5bGU9XCJtYXJnaW46IDQwcHhcIj5cbiAgICAgICAgICAgIDxhLWJ1dHRvbiB0eXBlPVwicHJpbWFyeVwiPkFmZml4IGluIHNjcm9sbGluZyBjb250YWluZXI8L2EtYnV0dG9uPlxuICAgICAgICAgIDwvYS1hZmZpeD5cbiAgICAgICAgPC9kaXY+XG4gICAgICA8L2Rpdj5cbiAgICA8L2Rpdj5cbiAgPC9hLW1vZGFsPlxuPC90ZW1wbGF0ZT5cblxuPHN0eWxlPjwvc3R5bGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)

### ref

- [modal-box](https://arco.design/vue/component/modal#width)
- [affix-container](https://arco.design/vue/component/affix#container)

**Reply by cdwmhcc** [reply]
> 

感谢回复。

示例中必须需要手动设置两个有高度的容器，这样的话滚动条会在 modal 的内部，modal 的自适应滚动会大大被限制。而 modal 本身是有滚动的，当 modal 的内容超过窗口的高度，在屏幕的最右侧它会自适应显示滚动条，如何绑定这个滚动。

``` vue
<template>
  <a-button @click="handleClick">Open Modal</a-button>
  <a-modal
    width="auto"
    v-model:visible="visible"
    @ok="handleOk"
    @cancel="handleCancel"
  >
    <template #title> Title </template>
    <div>
      <!-- 最外层需要设置一个有高度的容器 -->
      <div style="height: 200px; overflow: auto" ref="containerRef">
        <!-- 内层还需要设置一个有高度的容器 -->
        <div style="height: 400px; background: #cccccc; overflow: hidden">
          <a-affix :offsetTop="20" :target="containerRef" style="margin: 40px">
            <a-button type="primary">Affix in scrolling container</a-button>
          </a-affix>
        </div>
      </div>
    </div>
  </a-modal>
```

**Reply by hehehai** [reply]
嗯，大概明白你的意思了，目前 modal 还没有开放 body 的 ref。简单实现了下 [demo](https://arco-vue-playground.netlify.app/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgb25Nb3VudGVkIH0gZnJvbSAndnVlJ1xuXG5jb25zdCB2aXNpYmxlID0gcmVmKGZhbHNlKVxuXG5jb25zdCBoYW5kbGVDbGljayA9ICgpID0+IHtcbiAgdmlzaWJsZS52YWx1ZSA9IHRydWVcbn1cbmNvbnN0IGhhbmRsZU9rID0gKCkgPT4ge1xuICB2aXNpYmxlLnZhbHVlID0gZmFsc2Vcbn1cbmNvbnN0IGhhbmRsZUNhbmNlbCA9ICgpID0+IHtcbiAgdmlzaWJsZS52YWx1ZSA9IGZhbHNlXG59XG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8YS1idXR0b24gQGNsaWNrPVwiaGFuZGxlQ2xpY2tcIj5PcGVuIE1vZGFsPC9hLWJ1dHRvbj5cbiAgPGEtbW9kYWxcbiAgICA6d2lkdGg9XCI2MDBcIlxuICAgIHYtbW9kZWw6dmlzaWJsZT1cInZpc2libGVcIlxuICAgIEBvaz1cImhhbmRsZU9rXCJcbiAgICBAY2FuY2VsPVwiaGFuZGxlQ2FuY2VsXCJcbiAgICBib2R5LWNsYXNzPVwiYm9keS1jdXN0b21cIlxuICA+XG4gICAgPHRlbXBsYXRlICN0aXRsZT4gVGl0bGUgPC90ZW1wbGF0ZT5cbiAgICA8ZGl2IHN0eWxlPVwiaGVpZ2h0OiAxMjAwcHg7IGJhY2tncm91bmQ6ICNjY2NjY2M7IG92ZXJmbG93OiBoaWRkZW5cIj5cbiAgICAgIDxhLWFmZml4IDpvZmZzZXRUb3A9XCI0MFwiIHRhcmdldD1cIi5ib2R5LWN1c3RvbVwiIHN0eWxlPVwibWFyZ2luOiA0MHB4XCI+XG4gICAgICAgIDxhLWJ1dHRvbiB0eXBlPVwicHJpbWFyeVwiPkFmZml4IGluIHNjcm9sbGluZyBjb250YWluZXI8L2EtYnV0dG9uPlxuICAgICAgPC9hLWFmZml4PlxuICAgIDwvZGl2PlxuICA8L2EtbW9kYWw+XG48L3RlbXBsYXRlPlxuXG48c3R5bGU+XG4gIC5ib2R5LWN1c3RvbSB7XG4gICAgaGVpZ2h0OiA1MDBweDtcbiAgICBvdmVyZmxvdzogYXV0bztcbiAgfVxuPC9zdHlsZT5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==) 🤔 好像有闪烁出现

**Reply by cdwmhcc** [reply]
> 嗯，大概明白你的意思了，目前 modal 还没有开放 body 的 ref。简单实现了下 [demo](https://arco-vue-playground.netlify.app/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgb25Nb3VudGVkIH0gZnJvbSAndnVlJ1xuXG5jb25zdCB2aXNpYmxlID0gcmVmKGZhbHNlKVxuXG5jb25zdCBoYW5kbGVDbGljayA9ICgpID0+IHtcbiAgdmlzaWJsZS52YWx1ZSA9IHRydWVcbn1cbmNvbnN0IGhhbmRsZU9rID0gKCkgPT4ge1xuICB2aXNpYmxlLnZhbHVlID0gZmFsc2Vcbn1cbmNvbnN0IGhhbmRsZUNhbmNlbCA9ICgpID0+IHtcbiAgdmlzaWJsZS52YWx1ZSA9IGZhbHNlXG59XG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8YS1idXR0b24gQGNsaWNrPVwiaGFuZGxlQ2xpY2tcIj5PcGVuIE1vZGFsPC9hLWJ1dHRvbj5cbiAgPGEtbW9kYWxcbiAgICA6d2lkdGg9XCI2MDBcIlxuICAgIHYtbW9kZWw6dmlzaWJsZT1cInZpc2libGVcIlxuICAgIEBvaz1cImhhbmRsZU9rXCJcbiAgICBAY2FuY2VsPVwiaGFuZGxlQ2FuY2VsXCJcbiAgICBib2R5LWNsYXNzPVwiYm9keS1jdXN0b21cIlxuICA+XG4gICAgPHRlbXBsYXRlICN0aXRsZT4gVGl0bGUgPC90ZW1wbGF0ZT5cbiAgICA8ZGl2IHN0eWxlPVwiaGVpZ2h0OiAxMjAwcHg7IGJhY2tncm91bmQ6ICNjY2NjY2M7IG92ZXJmbG93OiBoaWRkZW5cIj5cbiAgICAgIDxhLWFmZml4IDpvZmZzZXRUb3A9XCI0MFwiIHRhcmdldD1cIi5ib2R5LWN1c3RvbVwiIHN0eWxlPVwibWFyZ2luOiA0MHB4XCI+XG4gICAgICAgIDxhLWJ1dHRvbiB0eXBlPVwicHJpbWFyeVwiPkFmZml4IGluIHNjcm9sbGluZyBjb250YWluZXI8L2EtYnV0dG9uPlxuICAgICAgPC9hLWFmZml4PlxuICAgIDwvZGl2PlxuICA8L2EtbW9kYWw+XG48L3RlbXBsYXRlPlxuXG48c3R5bGU+XG4gIC5ib2R5LWN1c3RvbSB7XG4gICAgaGVpZ2h0OiA1MDBweDtcbiAgICBvdmVyZmxvdzogYXV0bztcbiAgfVxuPC9zdHlsZT5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==) 🤔 好像有闪烁出现

![image](https://user-images.githubusercontent.com/8030134/177032819-6b346bbb-8b54-4ddf-a772-db7d29cbf29a.png)

感谢回复，我说的外层的滚动指的就是该滚动条在滚动时触发，而不是在 modal 内部再创建容器设定高度。

**Reply by hehehai** [reply]
你是指这样 [demo](https://arco-vue-playground.netlify.app/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgb25Nb3VudGVkIH0gZnJvbSAndnVlJ1xuXG5jb25zdCB2aXNpYmxlID0gcmVmKGZhbHNlKVxuXG5jb25zdCBoYW5kbGVDbGljayA9ICgpID0+IHtcbiAgdmlzaWJsZS52YWx1ZSA9IHRydWVcbn1cbmNvbnN0IGhhbmRsZU9rID0gKCkgPT4ge1xuICB2aXNpYmxlLnZhbHVlID0gZmFsc2Vcbn1cbmNvbnN0IGhhbmRsZUNhbmNlbCA9ICgpID0+IHtcbiAgdmlzaWJsZS52YWx1ZSA9IGZhbHNlXG59XG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8YS1idXR0b24gQGNsaWNrPVwiaGFuZGxlQ2xpY2tcIj5PcGVuIE1vZGFsPC9hLWJ1dHRvbj5cbiAgPGEtbW9kYWxcbiAgICBjbGFzcz1cIm1vZGFsLWN1c3RvbVwiXG4gICAgOndpZHRoPVwiNjAwXCJcbiAgICB2LW1vZGVsOnZpc2libGU9XCJ2aXNpYmxlXCJcbiAgICBAb2s9XCJoYW5kbGVPa1wiXG4gICAgQGNhbmNlbD1cImhhbmRsZUNhbmNlbFwiXG4gID5cbiAgICA8dGVtcGxhdGUgI3RpdGxlPiBUaXRsZSA8L3RlbXBsYXRlPlxuICAgIDxkaXYgc3R5bGU9XCJoZWlnaHQ6IDIyMDBweDsgYmFja2dyb3VuZDogI2NjY2NjYzsgb3ZlcmZsb3c6IGhpZGRlblwiPlxuICAgICAgPGEtYWZmaXggOm9mZnNldFRvcD1cIjQwXCIgdGFyZ2V0PVwiLm1vZGFsLWN1c3RvbSAuYXJjby1tb2RhbC13cmFwcGVyXCIgc3R5bGU9XCJtYXJnaW46IDQwcHhcIj5cbiAgICAgICAgPGEtYnV0dG9uIHR5cGU9XCJwcmltYXJ5XCI+QWZmaXggaW4gc2Nyb2xsaW5nIGNvbnRhaW5lcjwvYS1idXR0b24+XG4gICAgICA8L2EtYWZmaXg+XG4gICAgPC9kaXY+XG4gIDwvYS1tb2RhbD5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)？

**Reply by cdwmhcc** [reply]
> 你是指这样 [demo](https://arco-vue-playground.netlify.app/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgb25Nb3VudGVkIH0gZnJvbSAndnVlJ1xuXG5jb25zdCB2aXNpYmxlID0gcmVmKGZhbHNlKVxuXG5jb25zdCBoYW5kbGVDbGljayA9ICgpID0+IHtcbiAgdmlzaWJsZS52YWx1ZSA9IHRydWVcbn1cbmNvbnN0IGhhbmRsZU9rID0gKCkgPT4ge1xuICB2aXNpYmxlLnZhbHVlID0gZmFsc2Vcbn1cbmNvbnN0IGhhbmRsZUNhbmNlbCA9ICgpID0+IHtcbiAgdmlzaWJsZS52YWx1ZSA9IGZhbHNlXG59XG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8YS1idXR0b24gQGNsaWNrPVwiaGFuZGxlQ2xpY2tcIj5PcGVuIE1vZGFsPC9hLWJ1dHRvbj5cbiAgPGEtbW9kYWxcbiAgICBjbGFzcz1cIm1vZGFsLWN1c3RvbVwiXG4gICAgOndpZHRoPVwiNjAwXCJcbiAgICB2LW1vZGVsOnZpc2libGU9XCJ2aXNpYmxlXCJcbiAgICBAb2s9XCJoYW5kbGVPa1wiXG4gICAgQGNhbmNlbD1cImhhbmRsZUNhbmNlbFwiXG4gID5cbiAgICA8dGVtcGxhdGUgI3RpdGxlPiBUaXRsZSA8L3RlbXBsYXRlPlxuICAgIDxkaXYgc3R5bGU9XCJoZWlnaHQ6IDIyMDBweDsgYmFja2dyb3VuZDogI2NjY2NjYzsgb3ZlcmZsb3c6IGhpZGRlblwiPlxuICAgICAgPGEtYWZmaXggOm9mZnNldFRvcD1cIjQwXCIgdGFyZ2V0PVwiLm1vZGFsLWN1c3RvbSAuYXJjby1tb2RhbC13cmFwcGVyXCIgc3R5bGU9XCJtYXJnaW46IDQwcHhcIj5cbiAgICAgICAgPGEtYnV0dG9uIHR5cGU9XCJwcmltYXJ5XCI+QWZmaXggaW4gc2Nyb2xsaW5nIGNvbnRhaW5lcjwvYS1idXR0b24+XG4gICAgICA8L2EtYWZmaXg+XG4gICAgPC9kaXY+XG4gIDwvYS1tb2RhbD5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)？

哇，非常感谢您的帮助以及示例，就是这个！
